cmake/SYCL.cmake

```diff
@@ -0,0 +1,60 @@
+#Macro to find a sycl implimentation
+#Called without arguments will try to find a compiler appropriate sycl
+#With one argument finds that specific implimentation
+#With 2 arguments is as above but second argument is treated like version
+#Not happy with this yet - 
+# - should be able to accept lists of implementaions and versions
+# - and target architectures CPU,GPU etc
+macro(find_sycl)
+    set(SYCL_FOUND FALSE )
+    if (${ARGC} EQUAL 0)
+        #If intel compiler try to find the intel 
+        if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+            #TODO IntelDPCPP is deprecated remove
+            list(APPEND candidates "IntelSYCL" "IntelDPCPP")
+            foreach(candidate IN LISTS candidates)
+                find_package(${candidate} QUIET)
+                if(${candidate}_FOUND)
+                    set(SYCL_FOUND TRUE )
+                    set(SYCL_IMPLEMENTATION ${candidate} )
+                    break()
+                endif()
+            endforeach()
+        else()
+            #not intel try one of these
+            list(APPEND candidates "AdaptiveCPP" "hipSYCL")
+            list(APPEND versions   ""            "0.9.2")
+            foreach(candidate version IN ZIP_LISTS candidates versions)
+                find_package(${candidate} ${version} QUIET)
+                if(${candidate}_FOUND)
+                    set(SYCL_FOUND TRUE )
+                    set(SYCL_IMPLEMENTATION ${candidate} )
+                    break()
+                endif()
+            endforeach()
```

Comment:
I think we do the least complicated thing we can get away with here. I think you keep this block and if none of these are found assume the cxx compiler is a sycl compiler.

There are too many permutations of  sycl implementations to try and handle them all here.

Reply:
Aside: I think we actually require 0.9.4 - it's been on the todo list to update.